shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/ByronTranslation.hs

```diff
@@ -0,0 +1,190 @@
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
```

Comment:
Why is this needed?

Reply:
For the constraint you mentioned above `ADDRHASH c ~ Blake2b_224`

Reply:
Oh :facepalm:. I thought it was maybe related to a redundant `HasCallStack` (I have seen such constraints in this repo before).

Reply:
(BTW, we have https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-consensus/src/Ouroboros/Consensus/Util/RedundantConstraints.hs for that in consensus, but not here of course)